libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationRoute.kt

```diff
@@ -0,0 +1,768 @@
+import com.mapbox.services.android.navigation.v5.navigation.NavigationRoute.Builder
+import com.mapbox.services.android.navigation.v5.utils.LocaleUtils
+import okhttp3.EventListener
+import okhttp3.Interceptor
+import retrofit2.Call
+import retrofit2.Callback
+
+/**
+ * The NavigationRoute class wraps the [MapboxDirections] class with parameters which
+ * <u>must</u> be set in order for a navigation session to successfully begin. While it is possible
+ * to pass in any [com.mapbox.api.directions.v5.models.DirectionsRoute] into
+ * [MapboxNavigation.startNavigation], using this class will ensure your
+ * request includes all the proper information needed for the navigation session to begin.
+ *
+ *
+ *
+ *
+ * Developer Note: MapboxDirections cannot be directly extended since it is an AutoValue class.
+ *
+ * 0.5.0
+ */
+private val EVENT_LISTENER = NavigationRouteEventListener()
+
+class NavigationRoute
+/**
+ * Package private constructor used for the [Builder.build] method.
+ *
+ * @param mapboxDirections a new instance of a [MapboxDirections] class
+ * @since 0.5.0
+ */
+internal constructor(private val mapboxDirections: MapboxDirections) {
+
+    /**
+     * Wrapper method for Retrofit's [Call.clone] call, useful for getting call information
+     * and allowing you to perform additional functions on this [NavigationRoute] class.
+     *
+     * @return cloned call
+     * @since 1.0.0
+     */
+    val call: Call<DirectionsResponse>
+        get() = mapboxDirections.cloneCall()
+
+    /**
+     * Call when you have constructed your navigation route with your desired parameters. A
+     * [Callback] must be passed into the method to handle both the response and failure.
+     *
+     * @param callback a RetroFit callback which contains an onResponse and onFailure
+     * @since 0.5.0
+     */
+    fun getRoute(callback: Callback<DirectionsResponse>) {
+        mapboxDirections.enqueueCall(NavigationRouteCallback(EVENT_LISTENER, callback))
+    }
+
+    /**
+     * Wrapper method for Retrofit's [Call.cancel] call, important to manually cancel call if
+     * the user dismisses the calling activity or no longer needs the returned results.
+     */
+    fun cancelCall() {
+        if (!call.isExecuted) {
+            call.cancel()
+        }
+    }
+
+    /**
+     * This builder is used to create a new request to the Mapbox Directions API and removes options
+     * which would cause this navigation SDK to not behave properly. At a bare minimum, your request
+     * must include an access token, an origin, and a destination. All other fields can be left alone
+     * inorder to use the default behaviour of the API.
+     *
+     *
+     * By default, the directions profile is set to driving with traffic but can be changed to
+     * reflect your users use-case.
+     *
+     *
+     * @since 0.5.0
+     */
+    class Builder internal constructor(private val directionsBuilder: MapboxDirections.Builder) {
```

Comment:
```suggestion
    class Builder internal constructor(private val directionsBuilder: MapboxDirections.Builder = MapboxDirections.builder()) {
```
and now might remove another constructor